File: Backend/src/main/java/com/aquatrack/service/PropertyAdminManagementService.java
package com.aquatrack.service;

import com.aquatrack.dto.propertyadminmanagement.PropertyAdminResponse;
import com.aquatrack.dto.propertyadminmanagement.PropertyAdminSummaryResponse;

import java.util.List;

public interface PropertyAdminManagementService {

    // ==========================================
    // Property Admin Management
    // ==========================================

    List<PropertyAdminSummaryResponse> getAllPropertyAdmins();

    PropertyAdminResponse getPropertyAdminById(
            Long propertyAdminId
    );

    void suspendPropertyAdmin(
            Long propertyAdminId
    );

    void reactivatePropertyAdmin(
            Long propertyAdminId
    );

}

File: Aquatrack-frontend/src/services/propertyAdminManagementService.js
import api from "../api/axios";

// ==========================================
// Get All Property Admins
// ==========================================

export const getAllPropertyAdmins = async () => {

    const response = await api.get(
        "/admin/property-admins"
    );

    return response.data;

};

// ==========================================
// Get Property Admin By Id
// ==========================================

export const getPropertyAdminById = async (id) => {

    const response = await api.get(
        `/admin/property-admins/${id}`
    );

    return response.data;

};

// ==========================================
// Suspend Property Admin
// ==========================================

export const suspendPropertyAdmin = async (id) => {

    const response = await api.put(
        `/admin/property-admins/${id}/suspend`
    );

    return response.data;

};

// ==========================================
// Reactivate Property Admin
// ==========================================

export const reactivatePropertyAdmin = async (id) => {

    const response = await api.put(
        `/admin/property-admins/${id}/reactivate`
    );

    return response.data;

};